route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -41,190 +41,210 @@ namespace route_following_plugin
         plugin_discovery_msg_.activated = true;
         plugin_discovery_msg_.type = cav_msgs::Plugin::STRATEGIC;
         plugin_discovery_msg_.capability = "strategic_plan/plan_maneuvers";
+
         pose_sub_ = nh_->subscribe("current_pose", 1, &RouteFollowingPlugin::pose_cb, this);
-        twist_sub_ = nh_->subscribe("current_velocity", 1, &RouteFollowingPlugin::twist_cd, this);
-        
-        pnh_->param<double>("minimal_maneuver_duration", mvr_duration_, 16.0);
-        pnh2_->param<double>("config_speed_limit",config_limit);
-        pnh_->param<double>("buffer_time_lanechange",buffer_lanechange_time_);
-        pnh2_->param<double>("guidance/route_end_jerk", jerk_, 1.0);
-        pnh2_->param<std::string>("guidance/lane_change_plugin", lane_change_plugin_);
+        twist_sub_ = nh_->subscribe("current_velocity", 1, &RouteFollowingPlugin::twist_cb, this);
+
+        // read ros parameters
+        pnh_->param<double>("minimal_maneuver_duration", min_mvr_duration_, 16.0);
+        pnh_->param<std::string>("lane_change_plugin", lane_change_plugin_);
+
         wml_.reset(new carma_wm::WMListener());
         // set world model point form wm listener
         wm_ = wml_->getWorldModel();
 
+        //set a route callback to update route and calculate maneuver
+        wml_->setRouteCallback([this]() { 
+
+            lanelet::BasicPoint2d current_loc(pose_msg_.pose.position.x, pose_msg_.pose.position.y);
+            this->latest_maneuver_plan_ = route_cb(wm_->getRoute()->shortestPath(), current_loc);
+        });
+        
         discovery_pub_timer_ = pnh_->createTimer(
             ros::Duration(ros::Rate(10.0)),
             [this](const auto&) { plugin_discovery_pub_.publish(plugin_discovery_msg_); });
     }
+
     void RouteFollowingPlugin::run()
     {
         initialize();
         ros::CARMANodeHandle::spin();
     }
-    bool RouteFollowingPlugin::plan_maneuver_cb(cav_srvs::PlanManeuversRequest &req, cav_srvs::PlanManeuversResponse &resp)
-    {        
-        lanelet::BasicPoint2d current_loc(pose_msg_.pose.position.x, pose_msg_.pose.position.y);
-        auto current_lanelets = lanelet::geometry::findNearest(wm_->getMap()->laneletLayer, current_loc, 10);       
-        if(current_lanelets.size() == 0)
+
+    void RouteFollowingPlugin::pose_cb(const geometry_msgs::PoseStampedConstPtr& msg)
+    {
+        pose_msg_ = geometry_msgs::PoseStamped(*msg.get());
```

Comment:
It would be cleaner to have this callback populate a member variable called current_loc_, rather than to have multiple other methods creating their local variables current_loc and populating them repeatedly.